libsolidity/codegen/LValue.cpp

```diff
@@ -152,12 +152,22 @@ ImmutableItem::ImmutableItem(CompilerContext& _compilerContext, VariableDeclarat
 	solAssert(_variable.immutable(), "");
 }
 
-void ImmutableItem::retrieveValue(SourceLocation const&, bool) const
+void ImmutableItem::retrieveValue(SourceLocation const&, bool _remove) const
 {
 	solUnimplementedAssert(m_dataType->isValueType(), "");
-	solAssert(!m_context.runtimeContext(), "Tried to read immutable at construction time.");
-	for (auto&& slotName: m_context.immutableVariableSlotNames(m_variable))
-		m_context.appendImmutable(slotName);
+
+	if (m_context.runtimeContext())
+	{
+		CompilerUtils utils(m_context);
+		m_context << m_context.immutableMemoryOffset(m_variable);
+		utils.loadFromMemoryDynamic(*m_dataType);
+
+		if (_remove)
+			m_context << Instruction::POP;
```

Comment:
I don't think that's what ``_remove`` means :-). Compare with ``MemoryItem::retrieveValue`` - ``_remove`` means to remove the reference to the lvalue from stack, not the result of fetching it. Since immutables don't have references, this won't apply to them.

Reply:
Ah, I was looking at the others and I saw that the `loadFromMemoryDynamic` function mentioned memory_offset and length as post stack result so I thought that is what would be removed...